### ISSUE TITLE
db4o persistence not working in 1.8 RC

### ISSUE THREAD

**Issue by watou** [openhab/openhab#3728]
See discussion [here](https://community.openhab.org/t/db4o-persistence-broken-in-1-8/5889).  @teichsta  and @lewie?

**Comment by lewie** [comment]
@watou,
et voilà a quick Fix:

"org.openhab.persistence.db4o" was the "duplicated name".

In OSGI-INF/configuration.xml
changed 
````org.openhab.persistence.db4o```` to ````org.openhab.persistence.db4o.configuration````

In \OSGI-INF\database-db4o.xml
changed 
````xmlns:scr="http://www.osgi.org/xmlns/scr/v1.1.0"```` to ````xmlns:scr="http://www.osgi.org/xmlns/scr/v1.2.0"````

or use this Patch:

````
diff --git a/bundles/persistence/org.openhab.persistence.db4o/OSGI-INF/configuration.xml b/bundles/persistence/org.openhab.persistence.db4o/OSGI-INF/configuration.xml
index 2613baa..4f06c19 100644
--- a/bundles/persistence/org.openhab.persistence.db4o/OSGI-INF/configuration.xml
+++ b/bundles/persistence/org.openhab.persistence.db4o/OSGI-INF/configuration.xml
@@ -9,6 +9,6 @@
     http://www.eclipse.org/legal/epl-v10.html
 
 -->
-<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.2.0" activate="activate" name="org.openhab.persistence.db4o" immediate="true" configuration-pid="org.openhab.db4o" configuration-policy="optional">
+<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.2.0" activate="activate" name="org.openhab.persistence.db4o.configuration" immediate="true" configuration-pid="org.openhab.db4o" configuration-policy="optional">
    <implementation class="org.openhab.persistence.db4o.internal.Db4oConfiguration"/>
-</scr:component>
+</scr:component>
\ No newline at end of file
diff --git a/bundles/persistence/org.openhab.persistence.db4o/OSGI-INF/database-db4o.xml b/bundles/persistence/org.openhab.persistence.db4o/OSGI-INF/database-db4o.xml
index a4b41a9..857ffef 100644
--- a/bundles/persistence/org.openhab.persistence.db4o/OSGI-INF/database-db4o.xml
+++ b/bundles/persistence/org.openhab.persistence.db4o/OSGI-INF/database-db4o.xml
@@ -1,18 +1,18 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<!--
-
-    Copyright (c) 2010-2015, openHAB.org and others.
-
-    All rights reserved. This program and the accompanying materials
-    are made available under the terms of the Eclipse Public License v1.0
-    which accompanies this distribution, and is available at
-    http://www.eclipse.org/legal/epl-v10.html
-
--->
-<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.1.0" activate="activate" deactivate="deactivate" name="org.openhab.persistence.db4o">
-   <implementation class="org.openhab.persistence.db4o.internal.Db4oPersistenceService"/>
-   <service>
-      <provide interface="org.openhab.core.persistence.QueryablePersistenceService"/>
-      <provide interface="org.openhab.core.persistence.PersistenceService"/>
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright (c) 2010-2015, openHAB.org and others.
+
+    All rights reserved. This program and the accompanying materials
+    are made available under the terms of the Eclipse Public License v1.0
+    which accompanies this distribution, and is available at
+    http://www.eclipse.org/legal/epl-v10.html
+
+-->
+<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.2.0" activate="activate" deactivate="deactivate" name="org.openhab.persistence.db4o">
+   <implementation class="org.openhab.persistence.db4o.internal.Db4oPersistenceService"/>
+   <service>
+      <provide interface="org.openhab.core.persistence.QueryablePersistenceService"/>
+      <provide interface="org.openhab.core.persistence.PersistenceService"/>
    </service>
-</scr:component>
+</scr:component>

````
Thats it... :-)

Rgds
Herlmut

**Comment by watou** [comment]
Thanks, Helmut!  I will work up a PR and first ask the original reporters to try a new JAR.  I'm also removing the unneeded
```
      <provide interface="org.openhab.core.persistence.QueryablePersistenceService"/>
```
from `database-db4o.xml` and fixing two logging lines.

**Comment by watou** [comment]
[Test JAR](https://dl.dropboxusercontent.com/u/4286376/db4o-fixes/org.openhab.persistence.db4o-1.8.0-SNAPSHOT.jar)

**Comment by jdembski** [comment]
I can confirm that the db4o persistence is working with the JAR watou provided...

**Comment by teichsta** [comment]
fixed with #3738
